Sources/SwiftyStoreKit/ProductsInfoController.swift

```diff
@@ -69,9 +69,21 @@ class ProductsInfoController: NSObject {
             }
             inflightRequests[productIds] = InAppProductQuery(request: request, completionHandlers: [completion])
             request.start()
+
             return request
+
         } else {
+            
             inflightRequests[productIds]!.completionHandlers.append(completion)
+
+            let query = inflightRequests[productIds]!
+
+            if query.request.hasCompleted {
+                query.completionHandlers.forEach {
+                    $0(query.request.cachedResults!)
+                }
+            }
+
             return inflightRequests[productIds]!.request
```

Comment:
This section is the main part of this PR.
if the request has been completed, the completion closure that added won't be called.

Reply:
This causes an issue in purchasing operations.
If the developer retrieved a product before it. that operation would stop.